includes/Checker/Check_Runner.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Interface WordPress\Plugin_Check\Checker\Check_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Interface for runner classes like AJAX runner or CLI runner.
+ *
+ * @since n.e.x.t
+ */
+interface Check_Runner {
+
+	/**
+	 * Determine if the current request is intended for the plugin checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean Returns true if the check is for plugin else false.
+	 */
+	public function is_plugin_check();
+
+	/**
+	 * Handles running of the universal preparations depending on which checks requested.
+	 */
+	public function prepare();
+
+	/**
+	 * Run the requested checks against the plugin context and return the results.
+	 *
+	 * @return Check_Result Returns the check results.
```

Comment:
@vishalkakadiya again a minor edit here.

As well I believe the run method may also throw an exception if a Check fails. We will be calling the [`run_checks()`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Checks.php#L40-L47) method of the Checks class so take a look at that as an example.

```suggestion
	 * @return Check_Result Object containing all check results.
```